src/components/nav/Nav.js

```diff
@@ -1,11 +1,138 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import './Nav.scss';
+import NavModal from './NavModal';
 
 function Nav() {
-    return (
-        <div>
-            
+  const [onSearch, setOnSearch] = useState(false);
+  const [onMenu, setOnmenu] = useState(false);
+
+  const showSearchToggle = () => {
+    setOnSearch(Toggle => !Toggle);
+  };
+  const closeSearchModal = () => {
+    setOnSearch(false);
+  };
+
+  const menuModalToggle = () => {
+    setOnmenu(Toggle => !Toggle);
+  };
+
+  const navigate = useNavigate();
+
+  const goProduct = () => {
+    navigate('../product');
+  };
+  const goLogin = () => {
+    navigate('../login');
+  };
+
+  const goJoin = () => {
+    navigate('../join');
+  };
+
+  const goMypage = () => {
+    navigate('../mypage');
+  };
```

Comment:
```suggestion
  const goProduct = () => {
    navigate('/product');
  };
  const goLogin = () => {
    navigate('/login');
  };

  const goJoin = () => {
    navigate('/join');
  };

  const goMypage = () => {
    navigate('/mypage');
  };
```
그리고 이 부분도 위 리뷰 참고해주세요!